app/views/sitemap/page.xml.erb

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+    xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9
+        http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd
+        http://www.google.com/schemas/sitemap-image/1.1
+        http://www.google.com/schemas/sitemap-image/1.1/sitemap-image.xsd">
+    <% @topics.each do |topic| %>
+      <url>
+      <%-
+          url = "#{Discourse.base_url}/t/#{topic[1]}/#{topic[0]}"
+          if topic[4]
```

Comment:
this is too messy we got to clean it up... topic[4] is impossible to grok.

We can use DB.query and maintain field names or - since we are caching a lot of things just use `.select` ? not sure.... but something needs to be done here, plus logic belongs in helpers or controller